charts/jenkins/templates/cache-pvcs.yaml

```diff
@@ -1,30 +1,33 @@
-{{- /* create list of agents */}}
-{{- $_ := set .Values "agentList" (list .Values.agent) }}
+{{- /* create agent map */}}
+{{- /* merge .Values.agent into additional agent(s) to propogate values except cache.pvc.create */}}
+{{- $_ := set .Values "agentMap" (dict "agent" .Values.agent) }}
+{{- $agentCopy := deepCopy .Values.agentMap.agent }}
+{{- $_ := set $agentCopy.cache.pvc "create" false }}
 {{- range $name, $additionalAgent := .Values.additionalAgents }}
-    {{- /* merge .Values.agent into additional agent to ensure it at least has the default values */}}
-    {{- $_ := set $.Values "agentList" (append $.Values.agentList (merge $additionalAgent $.Values.agent)) }}
+    {{- /* use mergeOverwrite - see helm issue 7313 */}}
+    {{- $_ := set $.Values.agentMap $name (mergeOverwrite (deepCopy $agentCopy) $additionalAgent) }}
 {{- end }}
-{{- range $agent := .Values.agentList }}
+{{- range $name, $agent := .Values.agentMap }}
 {{- if $agent.cache.pvc.create }}
 ---
 apiVersion: v1
 kind: PersistentVolumeClaim
 metadata:
-  name: {{ tpl ($agent.cache.pvc.componentName | required "cache pvc componentName is required") $ }}
+  name: {{ include "jenkins.fullname" $ }}-{{ $agent.cache.pvc.componentName }}
   namespace: {{ template "jenkins.master.slaveKubernetesNamespace" $ }}

```

Comment:
```suggestion
  namespace: {{ template "jenkins.master.slaveKubernetesNamespace" . }}

```

Reply:
The `namespace` statement is failing with `.` when I test with minikube, maybe because of restricted scope in the `range` loop?